gateleen-core/src/main/java/org/swisspush/gateleen/core/util/StringUtils.java

```diff
@@ -211,11 +211,17 @@ public static String replaceWildcardConfigs(String contentWithWildcards,
      * @return the String with replaced wildcard values. Returns the input String when input String or
      * properties are <code>null</code>
      */
-    private static String replacePlainStringWildcardConfigs(String contentWithWildcards,
+    private static synchronized String replacePlainStringWildcardConfigs(String contentWithWildcards,
```

Comment:
Suggest to only synchronize the actual usage of the engine and not the whole method.

Synchronizing the whole method and then running into the early-return case (*props==null || conts==null*) 99% of the time would cost lots of performance due to useless locking.

Therefore I suggest to use somewhat like (*pseudCode*):
```java
if( fooOrBarIsNull ){
    return bar;
}
synchronized( StringUtils.class ){
    Engine engine = new Engine();
    return wildcardReplacementEngineIgnition(engine, contentWithWildcards, properties);
}
```

-> Same for the other method.

Reply:
OK, yes, makes sense, done